### PR TITLE
feat: add connectivity to transpile

### PR DESCRIPTION
**Pull request by shpface**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.

### Summary

This change will enable transpiling base on topology from #176

### CODE REVIEW

File: qiskit_braket_provider/providers/adapter.py [hunk 1]
```diff
@@ -427,10 +430,16 @@ def to_braket(
     basis_gates = set(basis_gates or _TRANSLATABLE_QISKIT_GATE_NAMES)
 
     braket_circuit = Circuit()
-    if not verbatim and not {gate.name for gate, _, _ in circuit.data}.issubset(
-        basis_gates
+    if not verbatim and (
+        not {gate.name for gate, _, _ in circuit.data}.issubset(basis_gates)
+        or connectivity
     ):
```
Comment (rmshaffer) [inline qiskit_braket_provider/providers/adapter.py:436]: Consider refactoring for readability, perhaps:
```suggestion
    uses_only_basis_gates = {gate.name for gate, _, _ in circuit.data}.issubset(basis_gates)
    needs_transpilation = connectivity or not uses_only_basis_gates
    if not verbatim and needs_transpilation:
```

File: tests/providers/test_adapter.py [hunk 2]
```diff
@@ -515,6 +515,43 @@ def test_get_controlled_gateset(self):
...
+        qiskit_circuit = QuantumCircuit(3)
+        qiskit_circuit.h(0)
+        qiskit_circuit.cx(0, 1)
+        qiskit_circuit.rxx(0.1, 0, 2)
+        connectivity = [[0, 1], [1, 0], [1, 2], [2, 1]]
+
+        braket_circuit = to_braket(
+            qiskit_circuit, connectivity=[[0, 1], [1, 0], [1, 2], [2, 1]]
+        )
+        braket_circuit_unconnected = to_braket(qiskit_circuit)
```
Comment (rmshaffer) [inline tests/providers/test_adapter.py:529]: Can you also add a test where `verbatim=True` to ensure that `connectivity` is ignored?

File: tests/providers/test_adapter.py [hunk 3]
```diff
@@ -515,6 +515,43 @@ def test_get_controlled_gateset(self):
...
+
+        assert all(
+            (
+                any(
+                    (
+                        gate.target.union(gate.control).issubset(adjacency)
+                        for adjacency in connectivity
+                    )
+                )
```
Comment (rmshaffer) [inline tests/providers/test_adapter.py:538]: For readability you could consider breaking this out into a helper function like
```
def is_connected(gate):
    return any(
        (gate.target.union(gate.control).issubset(adjacency) for adjacency in connectivity)
    )
```
and then these two assertions become just
```
assert all((is_connected(gate) for gate in braket_circuit.instructions))
assert not all((is_connected(gate) for gate in braket_circuit_unconnected.instructions))
```

**Review by rmshaffer — DISMISSED** [review]
Looks good! Left a couple of suggestions for readability.

**Comment by shpface** [comment]
> Looks good! Left a couple of suggestions for readability.

Thanks for the comments, Ryan

**Review by rmshaffer — APPROVED** [review]

**Pull request merged**